src/types/message-types.ts

```diff
@@ -70,3 +70,22 @@ export type RangeFilter = (GT | LT) & Partial<GT> & Partial<LT>;
 export type Filter = {
   [property: string]: EqualFilter | OneOfFilter | RangeFilter
 };
+
+export type Pagination = {
+  offset?: number
+  limit?: number
+};
+
+export enum RecordsDateSort {
+  CreatedAscending = 'createdAscending',
+  CreatedDescending = 'createdDescending',
+  PublishedAscending = 'publishedAscending',
+  PublishedDescending = 'publishedDescending',
+}
+
+export enum TimestampDateSort {
```

Comment:
Unless there is requirement for exposing this, I think we can skip.

Reply:
I think you're right, especially given the general abstraction you mentioned below for sort.